### PR TITLE
best-practices: add notebooks_unit_tests and more

### PR DESCRIPTION
**Pull request by bethz**
### Description
1. Initial release of notebooks_unit_tests
2. Add query to obtain GitHub build branch, include branch in submission to aml, then tag aml run with the branch.
3. Remove/clean debug.logger statements
4. Updated docs https://github.com/Microsoft/Recommenders/wiki/TestStrategy

### Related Issues
New addition.

### Checklist:

- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [x] I have added tests covering my contributions.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: tests/ci/submit_azureml_pytest.py [hunk 1]
```diff
@@ -71,10 +71,10 @@ def setup_workspace(workspace_name, subscription_id, resource_group, cli_auth,
     Returns:
         ws: workspace reference
     """
-    logger.debug('setup: workspace_name is ', workspace_name)
-    logger.debug('setup: resource_group is ', resource_group)
-    logger.debug('setup: subid is ', subscription_id)
-    logger.debug('setup: location is ', location)
+    logger.debug('setup: workspace_name is {0}'.format(workspace_name))
```
Comment (gramhagen) [inline tests/ci/submit_azureml_pytest.py]: I suggest leveraging logging's [internal formatting](https://docs.python.org/3/library/logging.html#logging.Logger.debug) as it looks cleaner. e.g.:
```
logger.debug('setup: resource group is %s', resource_group)
```

File: tests/ci/run_pytest.py [hunk 2]
```diff
@@ -56,24 +61,20 @@ def create_arg_parser():
...
-    logger.debug("args.junitxml", args.xmlname)
-    logger.debug("junit=", "--junitxml="+args.xmlname)
-    logger.info('pytest run:',
-                ["pytest",
-                 args.testfolder,
-                 "-m",
-                 args.testmarkers,
-                 "--junitxml="+args.xmlname])
+    logger.debug('args.junitxml {0}'.format(args.xmlname))
+    logger.info('pytest run:  [pytest, {0} "-m" {1} "--junitxml={2}" ]'.format(
```
Comment (gramhagen) [inline tests/ci/run_pytest.py]: i would suggest building the execution string first then you can use it in the log and the subprocess.run call, that way they'll never get out of sync.

```
pytest_cmd = ['pytest', args.testfolder, '-m', args.testmarkers, '--junitxml={}'.format(args.xmlname)]

logger.info('pytest.run: {}'.format(' '.join(pytest_cmd))
subprocess.run(pytest_cmd)
```

Comment (bethz) [reply tests/ci/run_pytest.py]: done!

File: tests/ci/submit_azureml_pytest.py (hunk 1 above)
Comment (bethz) [reply tests/ci/submit_azureml_pytest.py]: agree with improved look. changed to aml norm of:
logger.debug('setup: bla'.format(arg))

thanks

**Pull request merged**